pkg/csi/service/vanilla/controller.go

```diff
@@ -209,6 +213,19 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
 			return err
 		}
 	}
+	// Go module to keep the metrics http server running all the time.
+	go func() {
+		prometheus.CsiInfo.WithLabelValues(version).Set(1)
+		for {
+			log.Info("Starting the http server to expose Prometheus metrics..")
```

Comment:
We should use a logger with a new context each time the server restarts.

Reply:
Can you explain the reasoning? I see other go modules are run that use the same log defined in the Init.

Reply:
I see we are using `ctx, log := logger.GetNewContextWithLogger()`, from the init so we are good.
I was concerned about ctx which can be canceled. but here we are using `Background` context which is never canceled so we are good.